### PR TITLE
Updates CatalogKindName API to return a resource instead of array

### PR DESCRIPTION
**Pull request by SM43**
This updates the API to return a single resource instead of an array
because in a catalog there won't be multiple resources of a kind
with the same name.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/design/resource.go [hunk 1]
```diff
@@ -161,47 +161,47 @@ var _ = Service("resource", func() {
...
+				Enum("task", "pipeline")
+			})
+			Attribute("name", String, "Name of resource", func() {
+				Example("name", "buildah")
+			})
+			Required("catalog", "kind", "name")
 		})
 		Result(Resource, func() {
-			View("default")
+			View("withoutVersion")
```
Comment (sthaha) [inline api/design/resource.go:200]: we should return the versions: [...]

File: api/pkg/service/resource/resource.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func (s *service) ByVersionID(ctx context.Context, p *resource.ByVersionIDPayloa
 }
 
 // find resources using name of catalog, resource name and kind of resource
-func (s *service) ByCatalogKindName(ctx context.Context, p *resource.ByCatalogKindNamePayload) (res resource.ResourceCollection, err error) {
+func (s *service) ByCatalogKindName(ctx context.Context, p *resource.ByCatalogKindNamePayload) (res *resource.Resource, err error) {
```
Comment (PuneetPunamiya) [inline api/pkg/service/resource/resource.go:173]: Do we need `res` here ? If m

Comment (PuneetPunamiya) [inline api/pkg/service/resource/resource.go:173]: Do we need `res` any more here ??

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve
/meow

**Comment by SM43** [comment]
Recreated PR as #116

**Pull request closed without merging**